cmd/describeCluster_test.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright © 2019 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package cmd
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDescribeCluster(t *testing.T) {
+
+	// Setup to run: px describe cluster and return the Stdout, Stderr,
+	// and a function to restore state
+	so, _, r := pxTestSetupCli("px describe cluster")
+
+	// Defer to cleanup state
+	defer r()
+
+	// Start the CLI
+	err := runPx()
+	assert.NoError(t, err)
+
+	lines := strings.Split(so.String(), "\n")
+	assert.Contains(t, lines, "Cluster ID: mock")
+	assert.Contains(t, lines, "Cluster Status: STATUS_OK")
+	assert.Contains(t, lines, "Version: 1.0.0-fake")
+	assert.Contains(t, lines, "Cluster UUID: ce6b309e289d620be08c964969aeb39f")
+	assert.Contains(t, lines[9], "node-1")
```

Comment:
These unit tests were done when we were just starting and only using the mock-sdk-server. Now that we can point at any cluster, we cannot depend on the values, we only test that we get in the output what px printed.

Reply:
I have removed one node restriction though, keeping check for at least one node to be present. If it is run on any other cluster other than mock server, we will not be able to verify the values because we wouldn't know what is the expected value.